spring-xd-test-fixtures/src/main/java/org/springframework/xd/test/fixtures/JdbcSink.java

```diff
@@ -71,12 +71,23 @@ public JdbcTemplate getJdbcTemplate() {
 	@Override
 	protected String toDSL() {
 		StringBuilder dsl = new StringBuilder();
+		Connection connection = null;
 		try {
-			dsl.append("jdbc --initializeDatabase="+initializeDB+" --url=" + dataSource.getConnection().getMetaData().getURL());
+			connection = dataSource.getConnection();
+			dsl.append("jdbc --initializeDatabase="+initializeDB+" --url=" + connection.getMetaData().getURL());
```

Comment:
Depending on the actual datasource we're using (this is used in tests, for the record) this, albeit "cleaner", may be problematic: what if we use a SingleConnectionDataSource. Should we use SmartDataSource or is it overkill?